olp-cpp-sdk-dataservice-read/src/VersionedLayerClient.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/dataservice/read/VersionedLayerClient.h>
+
+#include <olp/core/client/OlpClientFactory.h>
+#include <olp/core/thread/TaskScheduler.h>
+
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/context/Context.h>
+
+#include "generated/api/BlobApi.h"
+#include "generated/api/QueryApi.h"
+
+#include "olp/dataservice/read/Condition.h"
+
+#include "ApiClientLookup.h"
+#include "generated/model/Api.h"
+
+#include "repositories/ExecuteOrSchedule.inl"
+
+#include "olp/core/logging/Log.h"
+
+#include <algorithm>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+namespace {
+constexpr auto kLogTag = "VersionedLayerClient";
+}
+
+class VersionedLayerClient::Impl final {
```

Comment:
Should be in a separate file

Reply:
Interface class' implementation is too trivial, so spawning 2 separate files has minor to no benefit.

Reply:
Let's not rush and implement things as they become required.

Reply:
We don't need to rush but we also don't need to copy everything all the time. We already know that we will have more functionality so where is the rush? Also everywhere else we use separated impls.